signer/src/testing/transaction_signer.rs

```diff
@@ -331,20 +332,34 @@ where
         let test_data = self.generate_test_data(&mut rng);
         Self::write_test_data(&test_data, &mut handle.storage).await;
 
-        let coordinator_private_key = PrivateKey::new(&mut rng);
-
-        let transaction_sign_request = message::BitcoinTransactionSignRequest {
-            tx: testing::dummy::tx(&fake::Faker, &mut rng),
-            aggregate_key: dummy_aggregate_key,
-        };
-
         let chain_tip = handle
             .storage
             .get_bitcoin_canonical_chain_tip()
             .await
             .expect("storage failure")
             .expect("no chain tip");
 
+        let coordinator_public_key = transaction_coordinator::coordinator_public_key(
+            &chain_tip,
+            &signer_info.first().unwrap().signer_public_keys,
+        )
+        .expect("failed to compute coordinator public key")
+        .unwrap();
+
+        let coordinator_private_key = signer_info
+            .iter()
+            .find(|signer_info| {
+                PublicKey::from_private_key(&signer_info.signer_private_key)
+                    == coordinator_public_key
+            })
+            .unwrap()
+            .signer_private_key;
```

Comment:
It's probably worth making a proper type that wraps `Vec<SignerInfo>` so that we can have functions for these common operations. No need to do anything in the PR though.

Reply:
Makes sense. I can do it as a small follow-up if I find the time.